rusk/src/lib/chain/rusk.rs

```diff
@@ -108,9 +114,18 @@ impl Rusk {
         let mut event_hasher = Sha3_256::new();
 
         for unspent_tx in txs {
+            if started.elapsed() > timeout {
+                info!("execute_transactions timeout triggered {timeout:?}");
+                break;
+            }
             // Don't include transactions if migration is triggered otherwise
             // the session rollback will not re-execute migration
             if Some(block_height) == self.migration_height {
+                info!("Skipping transactions due to migration_height");
+                break;
+            }
+            if unspent_tx.inner.fee().gas_limit > block_gas_left {
+                info!("Skipping due gas_limit greater than left: {block_gas_left}");
```

Comment:
Please add the transaction ID to the log. We might need it in case of bug hunting

Reply:
Tx_id is computed at runtime
The idea was to not impact performance at all

However, since it's needed for the use case you mentioned, I would go for log every transactions together with their id